examples/tc_neighbor_sharing.py

```diff
@@ -0,0 +1,122 @@
+        ipr.tc("add", "ingress", wan_if["index"], "ffff:")
+        ipr.tc("add-filter", "bpf", wan_if["index"], ":1", fd=wan_fn.fd,
+               prio=1, name=wan_fn.name, parent="ffff:", action="drop",
+               classid=1, rate="128kbit", burst=1024 * 32, mtu=16 * 1024)
+        ipr.tc("add-filter", "bpf", wan_if["index"], ":2", fd=pass_fn.fd,
+               prio=2, name=pass_fn.name, parent="ffff:", action="drop",
+               classid=2, rate="1024kbit", burst=1024 * 32, mtu=16 * 1024)
+        self.wan_if = wan_if
+
+    # helper function to create a namespace and a veth connecting it
+    def _create_ns(self, name, ipaddr, fn):
+        ns_ipdb = IPDB(nl=NetNS(name))
+        ipdb.create(ifname="%sa" % name, kind="veth", peer="%sb" % name).commit()
+        with ipdb.interfaces["%sb" % name] as v:
+            v.net_ns_fd = ns_ipdb.nl.netns
+        with ipdb.interfaces["%sa" % name] as v:
+            v.up()
+        with ns_ipdb.interfaces["%sb" % name] as v:
+            v.ifname = "eth0"
+            v.add_ip("%s" % ipaddr)
+            v.up()
+        ifc = ipdb.interfaces["%sa" % name]
+        if fn:
+            ipr.tc("add", "ingress", ifc["index"], "ffff:")
+            ipr.tc("add-filter", "bpf", ifc["index"], ":1", fd=fn.fd, name=fn.name,
+                   parent="ffff:", action="ok", classid=1)
+        self.ipdbs.append(ns_ipdb)
+        self.namespaces.append(ns_ipdb.nl)
+        cmd = ["iperf", "-s", "-B", ipaddr.split("/")[0]]
+        self.processes.append(NSPopen(ns_ipdb.nl.netns, cmd))
+        return (ns_ipdb, ifc)
+
+    # start the namespaces that compose the network, interconnect them with the bridge,
+    # and attach the tc filters
+    def start(self):
+        neighbor_list = []
+        local_list = []
+        for i in range(0, num_neighbors):
+            neighbor_list.append(self._create_ns("neighbor%d" % i, "172.16.1.%d/24" % (i + 100), neighbor_fn))
+        for i in range(0, num_locals):
+            local_list.append(self._create_ns("local%d" % i, "172.16.1.%d/24" % (i + 150), pass_fn))
+
+        with ipdb.create(ifname="br100", kind="bridge") as br100:
+            for x in neighbor_list:
+                br100.add_port(x[1])
+            for x in local_list:
+                br100.add_port(x[1])
+            br100.add_port(self.wan_if)
+            br100.up()
+
+    def release(self):
+        for p in self.processes: p.kill(); p.release()
+        for db in self.ipdbs: db.release()
+        for ns in self.namespaces: ns.remove()
+
+try:
+    sim = SharedNetSimulation()
+    sim.start()
+    print("Network ready. Create a shell in the wan0 namespace and test with iperf")
+    print(" e.g.: ip netns exec wan0 iperf -t 2 -c 172.16.1.100")
+    input("Press enter when finished: ")
+finally:
+    if "sim" in locals(): sim.release()
+    if "br100" in ipdb.interfaces: ipdb.interfaces.br100.remove().commit()
+    sleep(2)
```

Comment:
@drzaeus77 — in the case when IPDB doesn't work well w/o `sleep()` syncs, pls report the bug, since the goal of `commit()` calls is to be reliably synchronous and to guarantee the system state when the `commit()` call is finished.